pkg/controller/linstornodeset/linstornodeset_controller.go

```diff
@@ -329,180 +334,283 @@ func (r *ReconcileLinstorNodeSet) reconcileResource(ctx context.Context, pns *pi
+	for range pods {
+		err := <-output
 		if err != nil {
-			compoundError = append(compoundError, err)
+			registerErrors = append(registerErrors, err)
 		}
 	}
 
-	return compoundError
+	return registerErrors
 }
 
-func (r *ReconcileLinstorNodeSet) reconcileSatNodeWithController(sat *piraeusv1alpha1.SatelliteStatus, pod corev1.Pod, ssl *piraeusv1alpha1.LinstorSSLConfig) error {
-	// Mark this true on successful exit from this function.
-	sat.RegisteredOnController = false
-
-	// TODO: Add Context w/ an infinite loop
-	if len(pod.Status.ContainerStatuses) != 0 && !pod.Status.ContainerStatuses[0].Ready {
-		return fmt.Errorf("NS reconcileSatNodeWithController: Nodeset pod %s is not ready, delaying registration on controller", pod.Spec.NodeName)
-	}
-
-	node, err := r.linstorClient.GetNodeOrCreate(context.TODO(), lapi.Node{
+func (r *ReconcileLinstorNodeSet) reconcileSingleNodeRegistration(ctx context.Context, nodeSet *piraeusv1alpha1.LinstorNodeSet, pod *corev1.Pod) error {
+	lNode, err := r.linstorClient.GetNodeOrCreate(ctx, lapi.Node{
 		Name: pod.Spec.NodeName,
 		Type: lc.Satellite,
 		NetInterfaces: []lapi.NetInterface{
 			{
 				Name:                    "default",
 				Address:                 pod.Status.HostIP,
-				SatellitePort:           ssl.Port(),
-				SatelliteEncryptionType: ssl.Type(),
+				SatellitePort:           nodeSet.Spec.SslConfig.Port(),
+				SatelliteEncryptionType: nodeSet.Spec.SslConfig.Type(),
 			},
 		},
 	})
 	if err != nil {
-		log.Debug("NS reconcileSatNodeWithController error")
 		return err
 	}
 
-	log.WithFields(logrus.Fields{
-		"nodeName":         node.Name,
-		"nodeType":         node.Type,
-		"connectionStatus": node.ConnectionStatus,
-	}).Debug("NS reconcileSatNodeWithController: Found / Added a Satellite Node")
-
-	sat.ConnectionStatus = node.ConnectionStatus
-	if sat.ConnectionStatus != lc.Online {
-		return fmt.Errorf("NS reconcileSatNodeWithController: waiting for node %s ConnectionStatus to be %s, current ConnectionStatus: %s",
-			pod.Spec.NodeName, lc.Online, sat.ConnectionStatus)
+	if lNode.ConnectionStatus != lc.Online {
+		return fmt.Errorf("node '%s' registered, but not online (%s)", lNode.Name, lNode.ConnectionStatus)
 	}
 
-	sat.RegisteredOnController = true
 	return nil
 }
 
-func (r *ReconcileLinstorNodeSet) reconcileStoragePoolsOnNode(sat *piraeusv1alpha1.SatelliteStatus, pools []piraeusv1alpha1.StoragePool, pod corev1.Pod) error {
+func (r *ReconcileLinstorNodeSet) reconcileAutomaticDeviceSetup(ctx context.Context, nodeSet *piraeusv1alpha1.LinstorNodeSet, pod *corev1.Pod) error {
+	logger := logrus.WithFields(logrus.Fields{
+		"Name":      nodeSet.Name,
+		"Namespace": nodeSet.Namespace,
+		"Pod":       pod.Name,
+		"Op":        "reconcileAutomaticDeviceSetup",
+	})
+	// No setup required
+	if nodeSet.Spec.AutomaticStorageSetupType == defaultAutomaticStorageSetupType {
+		return nil
+	}
+
+	logger.Warn("Automatic device set up enabled. This is a potentially dangerous operation!")
```

Comment:
Is it really that dangerous? Logging a warning for every node for every reconciliation run is very loud.